From 57b37c6f65e041c9340e9a5c8ffb72602f382f70 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 28 Jul 2020 18:55:21 +0200
Subject: [PATCH] timer: dw-apb: fix compiler warnings

readl() and writel() expect void *. Do not pass an integer value.

Remove unused include asm/arch/timer.h.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
Reviewed-by: Bin Meng <bin.meng@windriver.com>
---
 drivers/timer/dw-apb-timer.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/timer/dw-apb-timer.c b/drivers/timer/dw-apb-timer.c
index 68bc258131..799d5171ab 100644
--- a/drivers/timer/dw-apb-timer.c
+++ b/drivers/timer/dw-apb-timer.c
@@ -14,7 +14,6 @@
 #include <dm/device_compat.h>
 
 #include <asm/io.h>
-#include <asm/arch/timer.h>
 
 #define DW_APB_LOAD_VAL		0x0
 #define DW_APB_CURR_VAL		0x4
@@ -34,7 +33,7 @@ static u64 dw_apb_timer_get_count(struct udevice *dev)
 	 * requires the count to be incrementing. Invert the
 	 * result.
 	 */
-	return timer_conv_64(~readl(priv->regs + DW_APB_CURR_VAL));
+	return timer_conv_64(~readl((void *)(priv->regs + DW_APB_CURR_VAL)));
 }
 
 static int dw_apb_timer_probe(struct udevice *dev)
@@ -59,8 +58,8 @@ static int dw_apb_timer_probe(struct udevice *dev)
 	clk_free(&clk);
 
 	/* init timer */
-	writel(0xffffffff, priv->regs + DW_APB_LOAD_VAL);
-	writel(0xffffffff, priv->regs + DW_APB_CURR_VAL);
+	writel(0xffffffff, (void *)(priv->regs + DW_APB_LOAD_VAL));
+	writel(0xffffffff, (void *)(priv->regs + DW_APB_CURR_VAL));
 	setbits_le32(priv->regs + DW_APB_CTRL, 0x3);
 
 	return 0;
-- 
2.28.0

